File: Baekjoon/Greedy/24524.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
	string S, T;
	cin >> S >> T;

	vector<int> counter(T.size()); // i번째까지 완성된 문자 개수

	for (char s : S) {
		size_t idx = T.find(s);
		if (idx == string::npos) { // if s in T
			continue;
		}

		if (s == T[0]) {
			counter[0] += 1;
		} else if (idx != string::npos) {
			if (counter[idx - 1] > 0) {
				counter[idx - 1] -= 1;
				counter[idx] += 1;
			}
		}
	}

	cout << counter.back() << endl; // 마지막 요소

	return 0;
}

File: Baekjoon/Greedy/24524.py
S = input()
T = input()

counter = [0] * len(T) # i번쨰까지 완성된 문자 개수

for s in S:
	if s in T:
		if s == T[0]:
			counter[0] += 1
		else: # 첫번쨰 문자가 아닐 경우
			idx = T.find(s) # 없을 경우 -1 리턴
			if idx != -1:
				if counter[idx - 1] > 0:
					counter[idx - 1] -= 1
					counter[idx] += 1

print(counter[-1])
